plugins/scaffolder/src/components/fields/MultiEntityPicker/MultiEntityPicker.tsx

```diff
@@ -64,33 +65,17 @@ export const MultiEntityPicker = (props: MultiEntityPickerProps) => {
     uiSchema['ui:options']?.defaultNamespace || undefined;
 
   const catalogApi = useApi(catalogApiRef);
-
+  const entityPresentationApi = useApi(entityPresentationApiRef);
   const { value: entities, loading } = useAsync(async () => {
     const { items } = await catalogApi.getEntities(
       catalogFilter ? { filter: catalogFilter } : undefined,
     );
+
```

Comment:
Now that https://github.com/backstage/backstage/pull/24595 is merged, could you instead do like

```ts
const entityRefToPresentation = new Map(
  await Promise.all(
    items.map(async item => {
      return [
        stringifyEntityRef(item),
        await entityPresentationApi.forEntity(item).promise
      ]
    })
  )
)

return { entities, entityRefToPresentation };
```

and return that together with the items?

Then you have a mapping from entity ref to presentation, synchronously available without having to worry about async, in all code below immediately when loading finishes